src/mono/mono.proj

```diff
@@ -153,29 +187,80 @@
-      <ActivateCmd Condition="'$(HostOS)' == 'windows'">powershell -NonInteractive -command &quot;&amp; $(ActivateCmd); Exit $LastExitCode &quot;</ActivateCmd>
-      <PythonCmd Condition="'$(HostOS)' == 'windows'and '$(TargetsBrowser)' == 'true'">setlocal EnableDelayedExpansion &amp;&amp; call &quot;$([MSBuild]::NormalizePath('$(EMSDK_PATH)', 'emsdk_env.bat'))&quot; &amp;&amp; !EMSDK_PYTHON!</PythonCmd>
+      <_EmsdkPaths Condition="'$(HostOS)' != 'windows'">
+export EMSDK_PATH=$(EMSDK_PATH)
+export DOTNET_EMSCRIPTEN_LLVM_ROOT=$(EMSDK_PATH)bin/
+export DOTNET_EMSCRIPTEN_NODE_JS=$(EMSDK_PATH)node/bin/node
+export DOTNET_EMSCRIPTEN_BINARYEN_ROOT=$(EMSDK_PATH)
+
+      </_EmsdkPaths>
+      <_EmsdkPaths Condition="'$(HostOS)' == 'windows'">
+set EMSDK_PATH=$(EMSDK_PATH)
+set EMSDK_PYTHON=$(EMSDK_PATH)python/python.exe
+set DOTNET_EMSCRIPTEN_LLVM_ROOT=$(EMSDK_PATH)bin/
+set DOTNET_EMSCRIPTEN_NODE_JS=$(EMSDK_PATH)node/bin/node
+set DOTNET_EMSCRIPTEN_NODE_PATH=$(EMSDK_PATH)node/bin/
+set DOTNET_EMSCRIPTEN_BINARYEN_ROOT=$(EMSDK_PATH)
+
+      </_EmsdkPaths>
+      <_EmscriptenPaths>
+import os
+
+LLVM_ROOT = os.path.expanduser('$(EMSDK_PATH)bin/')
+NODE_JS = os.path.expanduser('$(EMSDK_PATH)node/bin/node')
+BINARYEN_ROOT = os.path.expanduser('$(EMSDK_PATH)')
+
+FROZEN_CACHE = bool(os.getenv('FROZEN_CACHE', 'True'))
+
+COMPILER_ENGINE = NODE_JS
+JS_ENGINES = [NODE_JS]
+      </_EmscriptenPaths>
+      <PythonCmd Condition="'$(HostOS)' == 'windows'and '$(TargetsBrowser)' == 'true'">setlocal EnableDelayedExpansion &amp;&amp; call &quot;$([MSBuild]::NormalizePath('$(EMSDK_PATH)', 'emsdk_env$(ScriptExt)'))&quot; &amp;&amp; !EMSDK_PYTHON!</PythonCmd>
     </PropertyGroup>
 
     <RemoveDir Directories="$(EMSDK_PATH)" />
-    <Exec Command="git clone https://github.com/emscripten-core/emsdk.git emsdk"
-          WorkingDirectory="$(BrowserLocalPath)"
-          IgnoreStandardErrorWarningFormat="true" />
-    <Exec Command="git checkout $(EmscriptenVersion) &amp;&amp; $(InstallCmd) &amp;&amp; $(ActivateCmd)"
-          WorkingDirectory="$(EMSDK_PATH)"
-          IgnoreStandardErrorWarningFormat="true" />
-    <Exec Command="$(PythonCmd) -m pip install certifi &amp;&amp; $(PythonCmd) -m pip install pip-system-certs"
-          Condition="'$(HostOS)' == 'windows'"
-          WorkingDirectory="$(EMSDK_PATH)"
-          IgnoreStandardErrorWarningFormat="true" />
+    
+    <ItemGroup>
+      <EmsdkFiles   Condition="'%(PackageReference.Identity)' != 'runtime.$(_portableHostOS)-$(BuildArchitecture).Microsoft.NETCore.Runtime.Wasm.Node.Transport' and '%(PackageReference.Identity)' != 'Microsoft.NET.Runtime.Emscripten.3.1.34.Python.win-$(BuildArchitecture)'"
+                    Include="$(NuGetPackageRoot)\$([System.String]::Copy(%(PackageReference.Identity)).ToLowerInvariant())\%(PackageReference.Version)\tools\**" />
```

Comment:
I think instead of NuGetPackageRoot you should be able to use GeneratePackagePath

Reply:
I tried that and it gets complicated as the package names are based on other properties, so the generated property name is for example `PkgMicrosoft_NET_Runtime_Emscripten_3_1_34_Cache_$(_portableHostOS)-$(BuildArchitecture)`. It would work much better if the package path was added to the `PackageReference` items instead.